platform_data/defs.bzl

```diff
@@ -0,0 +1,86 @@
+"""
+The platform_data rule can be used to change the target platform of a target,
+and then depend on that elsewhere in the build tree. For example:
+
+cc_binary(name = "foo")
+
+platform_data(
+    name = "foo_embedded",
+    target = ":foo",
+    platform = "//my/new:platform",
+)
+
+py_binary(
+    name = "flasher",
+    srcs = ...,
+    data = [
+        ":foo_embedded",
+    ],
+)
+
+Regardless of what platform the top-level :flasher binary is built for,
+the :foo_embedded target will be built for //my/new:platform. 
+
+Note that if you depend on :foo_embedded it's not exactly the same as depending on :foo, since it won't forward all the same providers. In the future, we can extend this to add some common providers as needed"""
+
+def _target_platform_transition_impl(attr):
+    return {
+        "//command_line_option:platforms": str(attr.platform),
+    }
+
+_target_platform_transition = transition(
+    implementation = _target_platform_transition_impl,
+    inputs = [],
```

Comment:
I'm not sure - we'd have to check. 

That's a performance optimization I'd not worry about unless we hear feedback it's a problem. I don't think we should worry about that here because it's not immediately clear if the Starlark alternative `return []` is any different in the implementation layer.